### PR TITLE
enchancement-243 Migration to bootstrap 5

### PR DESCRIPTION
**Pull request by mudassarlodhi**
implements #243  

**Why was this pull request needed?**
There was a requirement to upgrade the application from bootstrap version 4 to 5 that provides better UI experience as well as is independent of an additional library called jQuery.

**How this pull request solves it?**
The application was moved to bootstrap 5 version and necessary changes were done in order to make the existing UI components in the application work. The descriptive list of changes are below:
- In `package.json` bootstrap 4 , jquery, and popper.js packages were removed and bootstrap 5 package was added. 
- In `angular.json` scripts array reference to the file `jquery.min.js` was removed as bootstrap 5 is not dependent on it. Secondly `bootstrap.min.js` was also removed and in place of that `bootstrap.bundle.min.js` was added as a dependency. Lastly `bootstrap.min.css` was removed from styles array because the stylings import for the bootstrap are handled using sass so that we can do some customization. Refer to `styles.scss` to see import details.
- div elements with `input-group-prepend` were removed as they are no more required in input group elements. 
- Inline forms are to be handled using grid layout using classes `row`, `col` and related classes. `form-inline` class has no effect on form tags/elements.
- `table-light` doesnt color the table header _gray_. this class was replaced with `table-secondary`. 
- margin classes `ml` and `mr` were replaced with bootstrap 5 supported `ms` and `me` classes.
- `styles.scss` has import for the bootstrap stylings scss file. some variables were overridden so that our application looks somewhat similar to what it looked like when using bootstrap 4. Default paddings for some components like **card** and **navbar** were quite less than they were in bootstrap 4 resulting in our application appearing  a little odd.

**Note:** It is required to run command `npm install` to make bootstrap 5 implementation work in our application. If that doesn't work, try deleting `package-lock.json` before running `npm install`.

### CODE REVIEW

**Comment by mudassarlodhi** [comment]
@DavideViolante

File: client/styles.scss [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+$card-spacer-y: 1.25rem;
+$card-spacer-x: 1.25rem;
+$card-border-radius: 0.25rem;
+$card-cap-padding-x: 1.25rem;
+$card-cap-padding-y: 0.75rem;
+$navbar-padding-x: 1rem;
+
+@import "node_modules/bootstrap/scss/variables";
+@import "node_modules/bootstrap/scss/bootstrap";
+
```
Comment (DavideViolante) [inline client/styles.scss:14]:
```suggestion
// Import Bootstrap
@import '../node_modules/bootstrap/scss/functions';

// Add custom bootstrap styles here

@import '../node_modules/bootstrap/scss/variables';
@import '../node_modules/bootstrap/scss/bootstrap';

```

**Comment by DavideViolante** [comment]
Hello, thank you very much for this PR. It seems fine, I just removed custom Bootstrap code, it's ok for me if the app looks different as long as there is less custom code and as much as default code.

**Comment by mudassarlodhi** [comment]
> Hello, thank you very much for this PR. It seems fine, I just removed custom Bootstrap code, it's ok for me if the app looks different as long as there is less custom code and as much as default code.

no problem. i think with that custom code removed. you don't need to import `functions.scss` and `variables.scss`. just the `bootstrap.scss` will be required. or you can removed that as well and include `bootstrap.min.css` in angular.json's styles array.

**Comment by DavideViolante** [comment]
Thank you, I removed unused imports and [added](https://github.com/DavideViolante/Angular-Full-Stack/commit/c51c489ece3cd9c7760e3936bd6e0940cfcdefcc) an example how to override bootstrap, it's useful to know.

**Pull request merged**